modules/fedimint-wallet-server/src/lib.rs

```diff
@@ -780,6 +789,11 @@ impl ServerModule for Wallet {
         dbtx.insert_new_entry(&PegOutTxSignatureCI(txid), &sigs)
             .await;
 
+        let propose_citem_tx = self.propose_citem.clone();
+        dbtx.on_commit(move || {
+            propose_citem_tx.notify_one();
```

Comment:
While technically there should be only one waiter instance, it's easy to mess up and it would cost us very little to just call [`notify_waiters`](https://docs.rs/tokio/latest/tokio/sync/struct.Notify.html#method.notify_waiters) instead.